csharp/src/Apache.Arrow/C/CArrowArrayExporter.cs

```diff
@@ -15,16 +15,21 @@
 
 
 using System;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using Apache.Arrow.Memory;
 
 namespace Apache.Arrow.C
 {
     public static class CArrowArrayExporter
     {
+#if NET5_0_OR_GREATER
```

Comment:
Perhaps add a comment explaining why this needs .Net 5.0+? Or will it be obvious to a C# developer?

Reply:
`UnmanagedCallersOnlyAttribute` was introduced in .NET 5.